### PR TITLE
feat: neon v1.15

### PR DESCRIPTION
**Pull request by 00nktk**

### CODE REVIEW

File: executor/Cargo.toml [hunk 1]
```diff
@@ -21,15 +21,17 @@ tokio.workspace = true
...
 rust_decimal_macros = "1.35"
 serde_json = "1"
 clap = { version = "4.5.4", features = ["derive"] }
 hex-literal = "0.4.1"
+semver = "1.0.23"
+arc-swap = "1.7.1"
 
 [dev-dependencies]
 serial_test = "3.1.1"
+jsonrpsee-core = "0.22"
```
Comment (polachok) [inline executor/Cargo.toml:34]: is this only for async trait? can't we just import it directly

File: executor/src/transactions.rs [hunk 2]
```diff
@@ -277,19 +328,95 @@ impl TransactionBuilder {
...
         chain_id: u64,
         alt: Option<AltInfo>,
     ) -> anyhow::Result<OngoingTransaction> {
-        let fallback_iterative =
-            |tx_data, alt| async move { self.empty_holder_for_iterative_data(tx_data, alt).await };
+        // NOTE: Use the last version with the old behaviour to support pre release version.
+        // (e.g. 1.0.0-dev < 1.0.0)
+        const BREAKPOINT: Version = Version::new(1, 14, 5);
+
+        if self.version > BREAKPOINT {
```
Comment (polachok) [inline executor/src/transactions.rs:346]: but what if they release 1.14.6 🤔  ?

File: executor/Cargo.toml (hunk 1 above)
Comment (00nktk) [reply executor/Cargo.toml]: don

File: executor/src/transactions.rs (hunk 2 above)
Comment (00nktk) [reply executor/src/transactions.rs:346]: how about `1.14.u64::MAX`?

Comment (polachok) [reply executor/src/transactions.rs:346]: why not >= 1.15 ?

Comment (00nktk) [reply executor/src/transactions.rs:346]: `1.15.0-dev < 1.15.0` see [this](https://docs.rs/semver/latest/semver/struct.Version.html#total-ordering)

**Review by polachok — APPROVED** [review]

**Pull request merged**